src/mawpy/utilities/common.py

```diff
@@ -47,6 +60,14 @@ def get_combined_stay(df_by_user: pd.DataFrame, threshold: float = 0.2) -> pd.Da
                         and distance between them is less than threshold, then this_stay, next_stay, next_to_next_stay
                          are merged.
     """
+    # Validate DataFrames
+    if df_by_user.empty:
+        raise ValueError("DataFrame (df_by_user) empty; cannot perform merge.")
+
+    # Check for required columns
+    required_columns = [STAY, STAY_LAT_LONG]
```

Comment:
We may not require this check for the same reasons above. On a separate note - STAY_LAT_LONG = ["stay_lat", "stay_long"] that makes required columns as ["stay", ["stay_lat", "stay_long"]]. This may be one the reasons for test failure.